lighthouse-core/runner.js

```diff
@@ -134,14 +136,21 @@ class Runner {
             a => Aggregate.aggregate(a, runResults.auditResults));
         }
 
+        let categories = [];
+        if (config.report) {
+          const reportGenerator = new ReportGeneratorV2();
+          categories = reportGenerator.generateReportJson(config, resultsById).categories;
+        }
+
         return {
           lighthouseVersion: require('../package').version,
           generatedTime: (new Date()).toJSON(),
           initialUrl: opts.initialUrl,
           url: opts.url,
-          audits: formattedAudits,
+          audits: resultsById,
           artifacts: runResults.artifacts,
           runtimeConfig: Runner.getRuntimeConfig(opts.flags),
+          categories,
```

Comment:
Yeah I'm in favor of restructuring this object for v2, should probably add a section to the doc to discuss format

I was thinking something like the following?
```
{
  report: {
    meta: {version, time, url}, // stuff thats currently top-level
    config, // full config used (after extension, cli overrides, etc)
    score, // final overall score
    categories, // heirarchy of computed scores/results
    audits // individual results by audit id
  } // the stuff that gets written to report.json and whose contents can be used to render the HTML report without any additional information
  artifacts: {gatherers, trace, perfLog/devtoolsMessages} // used to quickly re-run analysis/perform later deep dives/debug/etc
}
```